### PR TITLE
New Feature, StorageFilesystemExternal allows for filesystem function…

### PR DESCRIPTION
**Pull request by madsi1m**
…s to be run by an external script, rather than by php itself

### CODE REVIEW

File: classes/storage/StorageFilesystemExternal.class.php [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    public static function readChunk(File $file, $offset, $length) {
+
+	if ($file->transfer->options['encryption'])
+	    $offset=$offset/Config::get('upload_chunk_size')*Config::get('upload_crypted_chunk_size');
+
+	$out=self::run('fs_readChunk "'.$file->uid.'" '.$offset.' '.$length);
+	if ($out['status']!=0) {
+		//something bad
+        	//if(!file_exists($chunkFile))
+	        //    throw new StorageFilesystemFileNotFoundException($file_path, $file);
```
Comment (jornane) [inline classes/storage/StorageFilesystemExternal.class.php]: Shouldn't this cause an error?

Comment (madsi1m) [reply classes/storage/StorageFilesystemExternal.class.php]: You are right, will fix

**Comment by monkeyiq** [comment]
It might be nice for the StorageFilesystemExternalStream::run() to be removed and the StorageFilesystemExternal::run called in it's place. These two runs seem the same right now.

**Comment by monkeyiq** [comment]
I think I'll bring this in with the run() merge as a post option.

**Pull request merged**